src/frontend/js/components/RootSearchSuggestField/index.tsx

```diff
@@ -0,0 +1,174 @@
+ */
+export const RootSearchSuggestField = ({
+  courseSearchPageUrl,
+}: RootSearchSuggestFieldProps) => {
+  const intl = useIntl();
+
+  // Initialize hooks for the two pieces of state the controlled <Autosuggest> component needs to interact with:
+  // the current list of suggestions and the input value.
+  // Note: the input value is initialized from the courseSearchParams, ie. the `query` query string parameter
+  const [value, setValue] = useState('');
+  const [suggestions, setSuggestions] = useState<SearchSuggestionSection[]>([]);
+
+  // Helper to know if the user is currently highlighting a suggestion. This is necessary to make sure we do not
+  // override default `Autosuggest` behavior when we receive the ENTER keycode that triggers text search.
+  const [hasHighlightedSuggestion, setHasHighlightedSuggestion] = useState(
+    false,
+  );
+
+  const inputProps: SearchAutosuggestProps['inputProps'] = {
+    /**
+     * Callback triggered on every user input.
+     * @param _ Unused: change event.
+     * @param params Incoming parameters related to the change event.
+     * - `newValue` is the search suggest form field value.
+     */
+    onChange: (_, { newValue }) => {
+      // Always update the state
+      setValue(newValue);
+    },
+    onKeyDown: event => {
+      // When ther user presses enter from the search field, move to the course search view with
+      // whatever is currently in the field as a text query.
+      // Unless they are currently highlighting a suggestion, in which case we let Autosuggest handle it.
+      if (event.keyCode === 13 /* enter */ && !hasHighlightedSuggestion) {
+        location.replace(
+          `${courseSearchPageUrl}?${stringify({
+            ...API_LIST_DEFAULT_PARAMS,
+            query: value,
+          })}`,
+        );
+      }
+    },
+    placeholder: intl.formatMessage(messages.searchFieldPlaceholder),
+    value,
+  };
+
+  /**
+   * `react-autosuggest` callback triggered when the user picks a suggestion, to handle the interaction.
+   * Different interactions have different expected outcomes:
+   * - picking a course directs to the course detailed page (as this is a course search);
+   * - picking another resource suggestion adds that resource as a filter;
+   * - the default suggestion runs a full-text search on the content of the search suggest form field.
+   * @param _ Unused: selection event.
+   * @param suggestion `suggestion` as key to an anonymous object: the suggestion the user picked.
+   */
+  const onSuggestionSelected: SearchAutosuggestProps['onSuggestionSelected'] = async (
+    _,
+    { suggestion },
+  ) => {
+    // Course suggestions are treated differently: when the user finds a course they're interested in,
+    // we're just sending them to that's course page instead of the course search page.
+    if (isCourseSuggestion(suggestion)) {
+      return location.replace(suggestion.absolute_url);
```

Comment:
Do you think it's important to keep the session history ? If yes you should use `location.assign`.

Reply:
good catch!